### ISSUE TITLE
Minetest aborts after consequence of Lua error

### ISSUE THREAD

**Issue by kilbith** [minetest/minetest#10917]
Engine commit: 2072afb72b4b3e9c5dcbcec71d824aeae1b35d19

### Steps to reproduce:

1. Have a static Lua error in your mod
2. Try to play the game
3. Minetest fails gracefully
4. Fix the Lua code
5. Retry to play

```
terminate called after throwing an instance of 'BaseException*'
Thread 1 "minetest" received signal SIGABRT, Aborted.
```

Reported on the forums: https://forum.minetest.net/viewtopic.php?p=389517#p389517

**Comment by SmallJoker** [comment]
```
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007ffff6e5a859 in __GI_abort () at abort.c:79
#2  0x00007ffff722f951 in ?? () from /lib/x86_64-linux-gnu/libstdc++.so.6
#3  0x00007ffff723b47c in ?? () from /lib/x86_64-linux-gnu/libstdc++.so.6
#4  0x00007ffff723b4e7 in std::terminate() () from /lib/x86_64-linux-gnu/libstdc++.so.6
#5  0x00007ffff723b799 in __cxa_throw () from /lib/x86_64-linux-gnu/libstdc++.so.6
#6  0x000055555562c16d in Settings::createLayer(SettingsLayer, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) [clone .cold] ()
#7  0x0000555555b0dd24 in MapSettingsManager::MapSettingsManager(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) ()
#8  0x0000555555b03418 in ServerMap::ServerMap(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, IGameDef*, EmergeManager*, MetricsBackend*) ()
#9  0x0000555555b912ad in Server::init() ()
#10 0x0000555555b92241 in Server::start() ()
#11 0x00005555556c676b in Game::createSingleplayerServer(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, SubgameSpec const&, unsigned short) ()
#12 0x00005555556c7c68 in Game::startup(bool*, InputHandler*, GameStartData const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, bool*, ChatBackend*) ()
#13 0x00005555556d0208 in the_game(bool*, InputHandler*, GameStartData const&, std::__cxx11-----T--------------Ty--Type --Type <R--Type <--Typ----Ty--T--Type --Type <R--Type <--Type --Typ
``^

**Comment by SmallJoker** [comment]
Fixed the memory leak in d287da184
